org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserProjectFileParser.java

```diff
@@ -0,0 +1,39 @@
+package org.code.javabuilder;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class UserProjectFileParser {
+  private final ObjectMapper objectMapper;
+
+  public UserProjectFileParser() {
+    this.objectMapper = new ObjectMapper();
+  }
+
+  /**
+   * Parses json string containing file data and stores it in this.fileList.
+   *
+   * @param json JSON String in UserSourceData format
+   * @throws UserFacingException
+   * @throws UserInitiatedException
+   */
+  public List<ProjectFile> parseFileJson(String json)
```

Comment:
NIT: (up to you) This seems like it should be a static function since it's just doing a direct transform of data. Not sure how that would impact testing though.

Reply:
I think in general static functions are harder to test. For this case, the main reason it's not static is so we can share an ObjectMapper instance across multiple calls to parseFileJson

Reply:
👍